modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replication/request/ScanRetrieveBatchReplicaRequest.java

```diff
@@ -33,11 +35,52 @@ public interface ScanRetrieveBatchReplicaRequest extends ReplicaRequest {
     long scanId();
 
     /**
-     * Gets a scan row filter. The filter has a sense only for the first request, for the second one and the followings the field is
-     * ignored.
+     * Gets an index to use fot the retrieve request.
      *
-     * @return Row filter predicate.
+     * @return Index id.
      */
     @Marshallable
-    Predicate<BinaryRow> rowFilter();
+    UUID indexToUse();
+
+    /**
+     * Gets a key which is used for exact comparison in the index.
+     *
+     * @return Key to search.
+     */
+    @Marshallable
+    BinaryTuple exactKey();
+
+    /**
+     * Gets a lower bound to choose entries from {@link SortedIndexStorage}. Exclusivity is controlled by a {@link
+     * SortedIndexStorage#GREATER_OR_EQUAL} or {@link SortedIndexStorage#GREATER} flag. {@code null} means unbounded.
```

Comment:
Mentioning SortedIndexStorage is actually an abstraction leak. I'd expected that similar flags will be introduced in https://issues.apache.org/jira/browse/IGNITE-17748. Please add corresponding todo here, that'll denote that flag links should be changed.